### PR TITLE
feat: event log data model and hash chain (#13)

### LINKED ISSUE #13: Event log data model + hash chain
## Spec reference

- VISION.md Section 4.2 (The Event Log): Event format, hash chain, tamper-evident log
- VISION.md Section 4.8 (State Hashing): SHA-256, Merkle tree over state domains
- VISION.md Section 11 Decision 2: JSON serialization for events

## What to build

Define the Event data model and implement hash chain construction/verification in pure Nim. No storage backend — in-memory for testing.

### Event type

```
Event {
    sequence: u64,
    timestamp: DateTime,
    event_type: EventType,
    schema_version: u32,
    payload: bytes (JSON string),
    state_hash_after: [u8; 32] (SHA-256 hex string),
    parent_hash: [u8; 32] (SHA-256 hex string),
}
```

### Required functionality

1. **Event type definition** — Nim object with all fields from Section 4.2
2. **Event hashing** — SHA-256 of event content, producing `state_hash_after`
3. **Hash chain construction** — Each event's `parent_hash` = hash of previous event. First event's `parent_hash` = zeros.
4. **Hash chain verification** — Given a sequence of events, verify the chain is intact (each `parent_hash` matches previous event's hash)
5. **Event serialization** — JSON serialization/deserialization per Decision 2
6. **Event classification** — EventType enum: UserAction, ApiResponse, WebhookResult, CronResult, ProxyMinted

### Acceptance criteria

- [ ] Event type defined with all Section 4.2 fields
- [ ] Can append events to an in-memory log with automatic hash chain construction
- [ ] Can verify a valid hash chain (returns true)
- [ ] Can detect a tampered event in the chain (returns false + identifies which event)
- [ ] JSON round-trip: serialize → deserialize → identical event
- [ ] All tests pass

## Not in scope

- No storage backend (IndexedDB or SQLite) — that's issues #2 and #4
- No proxyFetch integration — that's issue #3
- No Merkle tree (Section 4.8) — hash chain only for now
- No snapshots or log compaction (Section 4.7) — future phase
- No fencing tokens (Section 4.3) — that's Phase 3

## Dependencies

None — this is the foundation for all other Phase 2 issues.

### PR DESCRIPTION
**Pull request by mikesol**
Closes #13

## What this does

Adds the `eventlog` module implementing the append-only event log data model with SHA-256 hash chain integrity. Defines the `Event` type, `EventLog` container with automatic hash chain construction, chain verification (`verifyChain`), delta extraction (`eventsSince`), and continuity verification for syncing (`verifyContinuity`). JSON serialization uses snake_case field names for wire format compatibility.

## Spec compliance

**VISION.md Section 4.2 (The Event Log):**
- Event fields match spec: `sequence`, `timestamp`, `event_type`, `schema_version`, `payload`, `state_hash_after`, `parent_hash`
- EventType enum covers all 5 spec types: `user_action`, `api_response`, `webhook_result`, `cron_result`, `proxy_minted`
- Hash chain: first event's `parent_hash` = 64 zeros, subsequent events chain via SHA-256 of canonical form
- `state_hash_after` computed by hashing event with `state_hash_after=""` (avoids circular dependency)
- Delta extraction via `eventsSince(sinceSequence)` for sync protocol
- `verifyContinuity(anchor, delta)` verifies deltas connect to known state
- SHA-256 via nimcrypto for native Nim; browser JS will use Web Crypto API (zero bundle cost) — implemented in #16

## Validation performed

- 28 tests passing (`nim c -r tests/test_eventlog.nim`)
- SHA-256 verified against known test vectors (empty string, "abc", "hello world")
- Hash chain integrity: tamper detection confirmed for payload, parentHash, and stateHashAfter modifications
- JSON round-trip serialization verified (serialize → deserialize → re-serialize = identical)
- Full `nimble test` passes (all 8 test suites including eventlog)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Introduced an event log system with create/append, JSON import/export, hashing, chain verification, and delta continuity checks
  * Event type parsing with validation

* **Public API**
  * Event log functionality is now exported from the main module surface

* **Tests**
  * Added comprehensive test suite covering behavior, edge cases, and tampering detection

* **Chores**
  * Added crypto dependency and updated CI to install dependencies and run tests

### CODE REVIEW

File: src/unanim/eventlog.nim [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  ## Serialize an Event to a JsonNode with snake_case field names.
+  result = %*{
+    "sequence": e.sequence.int,
+    "timestamp": e.timestamp,
+    "event_type": $e.eventType,
+    "schema_version": e.schemaVersion.int,
+    "payload": e.payload,
+    "state_hash_after": e.stateHashAfter,
+    "parent_hash": e.parentHash
+  }
```
[the remark this answers is not in the document]
Comment (mikesol) [reply src/unanim/eventlog.nim]: Dismissed — sequence numbers are append-only counters starting at 1. Reaching 2^63 events is not a realistic scenario. If it ever becomes one, we'd need a larger rethink than JSON serialization.

File: tests/test_eventlog.nim [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+    parentHash: "def456"
+  )
+  assert e.sequence == 1'u64
+  assert e.timestamp == "2026-02-09T12:00:00Z"
+  assert e.eventType == UserAction
+  assert e.schemaVersion == 1'u32
+  assert e.payload == """{"key": "value"}"""
+  assert e.stateHashAfter == "abc123"
+  assert e.parentHash == "def456"
+  echo "test_eventlog: Task 2b passed (Event construction)."
```
[the remark this answers is not in the document]
Comment (mikesol) [reply tests/test_eventlog.nim]: Dismissed — test suite is never run with `-d:release`. Both `nimble test` and CI use debug builds where `assert` is active.
